src/QueryablePromise.ts

```diff
@@ -38,7 +38,7 @@ class QueryablePromise {
   /**
    * Returns a resolved Promise and immediately mark it as 'done'
    */
-  static resolve (value) {
+  static resolve<T> (value: T): Promise<T> {
```

Comment:
Maybe not part of this PR, but shouldn't that also return a `QueryablePromise`? The properties of this returned `Promise` are writable, while the `QueryablePromise` properties only have getters.
And we have ternary operators that either return a `new QueryablePromise` or `QueryablePromise.resolve(...)`, so TypeScript could be more certain about the outcome there.

Reply:
Nevermind, as soon as we use `.then(...)` it reverts back to a normal `Promise`. Can we override `then` and `catch` in some way so the resulting Promise is also queryable, or is that a bad idea?

Reply:
You're right. `storeDate._meta.load` should be of type `QueryablePromise` and all methods modifying `load` should ensure it's of this type. Extending `Promise` and overriding `then`/`catch` could be an option. Need to give it a few thoughts.

Reply:
And `QueryablePromise.resolve` should also return `QueryablePromise` in that case